src/execution.c

```diff
@@ -69,19 +73,27 @@ static int	execute_cmds(t_macro *macro, int read_end)
 		if (pipe(macro->pipe_fd) == -1)
 			return (error_msg("pipe failed", i));
 		macro->pid[i] = fork();
-		if (macro->pid[i] == -1)
+		if (macro->pid[i] < 0)
 		{
 			close(macro->pipe_fd[1]);
 			return (error_msg("fork failed", i));
 		}
-		if (macro->pid[i] == 0)
-			execute_child_process(macro, i, read_end);
+		else if (macro->pid[i] == 0)
+			execute_child_process(macro, i, read_end, pipe_exit);
 		if (read_end > 0)
 			close(read_end);
 		read_end = macro->pipe_fd[0];
 		close(macro->pipe_fd[1]);
 		i++;
 	}
+			if (macro->pid != 0)
+		{
+			close (pipe_exit[1]);
```

Comment:
This chunk is repeated here and in the caller function. I removed it from here and it worked fine. I suggest closing the pipe in the caller function

Reply:
OK, I'll add it to the next release.